feat(lessons): add share code lessons to repo (#50)
pyOpenSci/lessons: publish-share-code/intro.md

# Publish & share your code

:::{toctree}
:hidden:
:caption: Share code
:maxdepth: 2

Intro <self>
Share your Code <share-code>
Cite your Code <cite-code>
:::

:::{toctree}
:hidden:
:caption: Activities 
:maxdepth: 2

Share your Code <activity-1-share-code>

:::

## Sharing and Citing Your Code

> Sharing your code using open, collaborative platforms is essential to open science because it adds transparency, encourages collaboration and reuse, and enables citability. 

Reviewer: ```suggestion
Sharing your code using open, collaborative platforms is essential to open science because it encourages collaboration and reuse, enables citability, and adds transparency needed for reproducibility. 
```

> There are many ways to share your code. Platforms like [GitHub](https://github.com) and [GitLab](https://gitlab.com) allow you to collaborate with others more efficiently and effectively. By sharing with the community, you can receive feedback and improve your projects through collective input.

Reviewer: ```suggestion
There are many ways to **share** your code. Platforms like [GitHub](https://github.com) and [GitLab](https://gitlab.com) allow you to collaborate with others efficiently and effectively. By sharing with the community, you can receive feedback and improve your projects through collective input.
```

> There are also many ways to make your code citable. You could opt to write a paper about your software and publish it in a journal. If you only want to publish your code, journals like the [Journal of Open Source Software (JOSS)](https://joss.theoj.org/) combined with peer-review processes like [pyOpenSci](https://www.pyopensci.org/) provide valuable support.

Reviewer: ```suggestion
There are also many ways to make your code **citable**. You could write a paper about your software and publish it in a journal. If you wish to publish your code only, journals like the [Journal of Open Source Software (JOSS)](https://joss.theoj.org/) combined with peer-review processes like [pyOpenSci](https://www.pyopensci.org/) provide valuable support.
```

> However, if you don't want to publish formally but still want your code to be citable, you can use tools like [Zenodo](https://zenodo.org/). Zenodo allows you to give proper credit to contributors and add credibility and visibility to your work without going through a formal publication process. Citing code encourages transparency and reproducibility, making it easier for others to build upon your efforts.

Reviewer: ```suggestion
If you don't want to publish formally but still want your code to be citable, you can use tools like [Zenodo](https://zenodo.org/). Zenodo allows you to give proper credit to contributors and add credibility to your work without going through a formal publication process. Citing code encourages transparency and reproducibility, making it easier for others to build upon your efforts.
```

> Finally, publishing your code reinforces open science principles and amplifies your impact by helping others solve problems and innovate more efficiently, ultimately advancing knowledge and progress in your field.

Reviewer: ```suggestion
Finally, **publishing** your code reinforces open science principles and amplifies your impact by helping others solve problems and innovate more efficiently, ultimately advancing knowledge and progress in your field.
```
